combine/combine.go

```diff
@@ -0,0 +1,426 @@
+	for i := 0; i < matchColumns; i++ {
+		if a[i] != b[i] {
+			return false
+		}
+	}
+
+	return true
+}
+
+// matchingSummaryLineMerger is a func used by mergeSummaryLines() to handle
+// summary columns when match columns match. a is the previous columns, b is the
+// current. a should have its summary columns altered to merge information from
+// b. Cols is the number of summary columns (the columns that contain info to
+// eg. sum).
+type matchingSummaryLineMerger func(cols int, a, b []string)
+
+// sumCountAndSize is a matchingSummaryLineMerger that, given cols 2,  will sum
+// the second to last element of a and b and store the result in a[penultimate],
+// and likewise for the last element in a[last]. This corresponds to summing the
+// file count and size columns of 2 lines in a by* file.
+func sumCountAndSize(cols int, a, b []string) {
+	last := len(a) - (cols - 1)
+	penultimate := last - 1
+
+	a[penultimate] = addNumberStrings(a[penultimate], b[penultimate])
+	a[last] = addNumberStrings(a[last], b[last])
+}
+
+// sumCountAndSizeAndKeepOldestAtime needs cols 3, and is like sumCountAndSize,
+// but keeps the oldest atime (smallest number) found in the last column.
+func sumCountAndSizeAndKeepOldestAtime(cols int, a, b []string) {
+	sumCountAndSize(cols, a, b)
+
+	last := len(a) - 1
+
+	if atoi(b[last]) < atoi(a[last]) {
+		a[last] = b[last]
+	}
+}
+
+// addNumberStrings treats a and b as ints, adds them together, and returns the
+// resulting int64 as a string.
+func addNumberStrings(a, b string) string {
+	return strconv.FormatInt(atoi(a)+atoi(b), intBase)
+}
+
+// atoi is like strconv.Atoi but returns an int64 and dies on error.
+func atoi(n string) int64 {
+	i, _ := strconv.ParseInt(n, intBase, 0)
+
+	return i
+}
+
+// mergeGroupStreamToFile merges pre-sorted (pre-merged) group data
+// (eg. from a `sort -m` of .bygroup files), summing consecutive lines with
+// the same first 2 columns, and outputting the results.
+func mergeGroupStreamToFile(data io.ReadCloser, output *os.File) error {
+	if err := MergeSummaryLines(data, groupSumCols, numSummaryColumns, sumCountAndSize, output); err != nil {
+		return err
+	}
+
+	return output.Close()
+}
+
+// MergeDGUTFilesToDB finds and merges the dgut files and then stores the
+// information in a database.
+func MergeDGUTFilesToDB(sourceDir string) error {
+	paths, err := fs.FindFilePathsInDir(sourceDir, statDGUTSummaryOutputFileSuffix)
+	if err != nil {
+		return err
+	}
+
+	outputDir, err := createCombineDGUTOutputDir(sourceDir)
+	if err != nil {
+		return err
+	}
+
+	err = mergeDGUTAndStoreInDB(paths, outputDir)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// createCombineDGUTOutputDir creates a dgut output dir in the given dir.
+// Returns the path to the created directory. If it already existed, will delete
+// it first, since we can't store to a pre-existing db.
+func createCombineDGUTOutputDir(dir string) (string, error) {
```

Comment:
Convert this to a general method in fs package that takes the full path to a directory to delete and then re-create.